test/unit/puppetlabs/services/jruby/jruby_puppet_agents_test.clj

```diff
@@ -98,4 +98,16 @@
             jruby-puppet
             jruby-service
             (is (instance? JRubyPuppet jruby-puppet))))
-        (is (= 4 @num-borrows))))))
\ No newline at end of file
+        (is (= 4 @num-borrows))))))
+
+(deftest next-instance-id-test
+  (let [pool-context (jruby-core/create-pool-context
+                       (jruby-testutils/jruby-puppet-config {:max-active-instances 8})
+                       jruby-testutils/default-profiler
+                       jruby-testutils/default-shutdown-fn)]
+    (testing "next instance id should be based on the pool size"
+      (is (= 10 (jruby-agents/next-instance-id 2 pool-context)))
+      (is (= 100 (jruby-agents/next-instance-id 92 pool-context))))
+    (testing "next instance id should wrap after max int"
+      (let [id (- Integer/MAX_VALUE 1)]
+        (is (= (mod id 8) (jruby-agents/next-instance-id id pool-context)))))))
```

Comment:
Okay, glad to hear that I misread the code and that the IDs increase by 1 and not `pool-size`.  

Disallowing negative IDs is fine with me.  However, choosing some arbitrary value to cap them at seems arbitrary and unnecessary.